src/backend/distributed/utils/node_metadata.c

```diff
@@ -1424,7 +1424,10 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 	{
 		Name nodeClusterName = DatumGetName(nodeCluster);
 		char *nodeClusterString = NameStr(*nodeClusterName);
-		strlcpy(workerNode->nodeCluster, nodeClusterString, NAMEDATALEN);
+		if (nodeClusterString != NULL)
```

Comment:
As long as I see, nodeClusterString shouldn't be NULL here.  master_initialize_node_metadata() call AddNodeMetadata() with nodeCluster = 'default'. And then in AddNodeMetadata(), we insert that row before we try to reach it.

```
	InsertNodeRow(nextNodeIdInt, nodeName, nodePort, groupId, nodeRack, hasMetadata,
				  isActive, nodeRole, nodeCluster);

	workerNode = FindWorkerNodeAnyCluster(nodeName, nodePort);
```